gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,383 @@
+    }
+});
+
+
+gpii.flowManager.getInstalledSolutions = function (that) {
+    return fluid.promise.fireTransformEvent(that.events.onSolutionsForCurrentDevice);
+};
+
+/**
+ * This main API entry point for capturing settings from a system or computer. This captures
+ * the actual settings on the device, so it assumed to be running in a local untrusted flow
+ * manager.
+ *
+ * @param {Event} onCaptureSettingsForCurrentDevice - The transforming promise chain
+ * @param {Object} options - Options for this chain.
+ * @param {Array} options.solutionsList - An array of solution IDs to filter by when
+ * retreiving settings. If this option is not included, all available settings will be
+ * returned. ex: `["com.microsoft.windows.mouseSettings", "com.freedomscientific.jaws"]`.
+ * @return {Promise} A promise resolved with the payload of captured system settings.
+ */
+gpii.flowManager.getSystemSettingsCapture = function (onCaptureSettingsForCurrentDevice, options) {
+    return fluid.promise.fireTransformEvent(onCaptureSettingsForCurrentDevice,
+        {}, options);
+};
+
+/*
+ * Runs through all the solutions currently available on the system, pulls the current
+ * setting for each supportedSetting and returns them in an object. Primary use case
+ * is for backing Capture tools that would allow a user to set up their GPII profile
+ * starting with the current settings for their applications on the local machine.
+ *
+ * @param {lifecycleManager.invokeSettingsHandlerGet} invokeSettingsHandlerGet - Invoker from the `lifecycleManager`
+ * necessary for extracting settings from a solution and it's spec.
+ * @param {lifecycleManager.variableResolver} resolver - Resolver for looking up handler specifications
+ * @param {lifecycleManager.userSession.localResolver} localResolver - Resolver for expanding configuration file paths
+ * from the solutions registry.
+ * @param {Object} solutions - Solutions registry entries for solutions available on the current machine.
+ * @param {Object} options - Extra options for processing.
+ * @param {Array} options.solutionsList - If provided, only solutions in this list of `solutionsID`s will
+ * be captures. Example:
+ *
+ * '''json
+ * ["com.microsoft.windows.cursors", "com.freedomscientific.jaws"]
+ * '''
+ * @return {fluid.promise} Returns a promise resolving with the entire system settings capture.
+ */
+gpii.flowManager.captureSystemSettings = function (invokeSettingsHandlerGet, resolver, localResolver, solutions, options) {
+    var promiseTogo = fluid.promise();
+
+    var handlerGetPromises = [];
+
+    var safeHandlerGet = function (invokeSettingsHandlerGet, solutionID, handlerSpec) {
+        var promiseTogo = fluid.promise();
+        try {
+            invokeSettingsHandlerGet(solutionID, handlerSpec).then(function (data) {
+                // Fix up the nativeWindows and SPI Settings Handlers
+                var filteredData = gpii.flowManager.filterInvokeSettingsHandlersReturn(data, handlerSpec);
```

Comment:
Same as the formatting, it probably makes more sense not to bind this filtering tightly with invokeSettingsHandlerGet(). It should be performed on the read result outside of the read() function.

Reply:
This has been cleaned up and removed as it's windows specific.